### PR TITLE
group by middleware

### PR DESCRIPTION
**Pull request by gkothe**
Hey, we needed the group by in our code here in our company.
Let me know if i need to modify anything or if you rather modify yourself. 
Thanks

### CODE REVIEW

**Comment by olivierwilkinson** [comment]
Hi there,

Thanks for doing this! The code and tests look good. I just want to look into the groupBy action not being included in NestedAction. If that was an oversight on my part it would be good to add it to the prisma-nested-middleware directly. This week has been crazy busy for me but I'll try to do that as soon as I can so this gets merged in good time.

File: src/lib/types.ts [hunk 1]
```diff
@@ -10,3 +11,5 @@ export type Config = {
   models: Partial<Record<Prisma.ModelName, ModelConfig | boolean>>;
   defaultConfig?: ModelConfig;
 };
+//extended like this so we didnt have to make a PR directly to prisma. NestedAction dont contains groupBy
+export type NestedCustomParams = Omit<NestedParams, "action"> & {action: NestedAction | "groupBy"}
```
Comment (olivierwilkinson) [inline src/lib/types.ts:15]: `NestedCustomParams` should no longer be needed 👍

Comment (gkothe) [reply src/lib/types.ts]: great

**Comment by gkothe** [comment]
all updated

**Review by olivierwilkinson — APPROVED** [review]
Looks great! 🙌

**Pull request merged**